core/src/main/java/io/questdb/griffin/engine/join/HashJoinLightRecordCursorFactory.java

```diff
@@ -187,13 +156,51 @@ public boolean hasNext() {
             return false;
         }
 
-        void of(RecordCursor masterCursor, RecordCursor slaveCursor) {
-            this.masterCursor = masterCursor;
-            this.slaveCursor = slaveCursor;
-            this.masterRecord = masterCursor.getRecord();
-            this.slaveRecord = slaveCursor.getRecordB();
-            record.of(masterRecord, slaveRecord);
-            slaveChainCursor = null;
+        void of(RecordCursor masterCursor, RecordCursor slaveCursor, SqlExecutionCircuitBreaker circuitBreaker) {
+            try {
+                this.isOpen = true;
```

Comment:
First time when you create Factory and then Cursor, maps are already inflated and `isOpen` is already true here. It's better to not re-inflate when `isOpen == true` here and all other similar places.

Reply:
I'll make the change but keep in mind that before memory is used it remains virtual, with no physical memory backing it up and as such reallocating to the same size should be a no-op.